packages/dd-trace/src/debugger/devtools_client/snapshot/processor.js

```diff
@@ -0,0 +1,241 @@
+      return toSet(obj.className, obj.properties, maxLength)
+    case 'weakmap':
+      return toMap(obj.className, obj.properties, maxLength)
+    case 'weakset':
+      return toSet(obj.className, obj.properties, maxLength)
+    // case 'iterator': // TODO: I've not been able to trigger this subtype
+    case 'generator':
+      // Use `subtype` instead of `className` to make it obvious it's a generator
+      return toObject(obj.subtype, obj.properties, maxLength)
+    case 'error':
+      // TODO: Convert stack trace to array to avoid string trunctation or disable truncation in this case?
+      return toObject(obj.className, obj.properties, maxLength)
+    case 'proxy':
+      // Use `desciption` instead of `className` as the `type` to get type of target object (`Proxy(Error)` vs `proxy`)
+      return toObject(obj.description, obj.properties, maxLength)
+    case 'promise':
+      return toObject(obj.className, obj.properties, maxLength)
+    case 'typedarray':
+      return toArray(obj.className, obj.properties, maxLength)
+    case 'arraybuffer':
+      return toArrayBuffer(obj.className, obj.properties, maxLength)
+    // case 'dataview': // TODO: Looks like the internal ArrayBuffer is only accessible via the `buffer` getter
+    // case 'webassemblymemory': // TODO: Looks like the internal ArrayBuffer is only accessible via the `buffer` getter
+    // case 'wasmvalue': // TODO: I've not been able to trigger this subtype
+    default:
+      // As of this writing, the Chrome DevTools Protocol doesn't allow any other subtypes than the ones listed above,
+      // but in the future new ones might be added.
+      return { type: obj.subtype, notCapturedReason: 'unknown' }
+  }
+}
+
+function toFunctionOrClass (value, maxLength) {
+  const classMatch = value.description.match(CLASS_REGEX)
+
+  if (classMatch === null) {
+    // This is a function
+    // TODO: Would it make sense to detect if it's an arrow function or not?
+    return toObject(value.className, value.properties, maxLength)
+  } else {
+    // This is a class
+    const className = classMatch[1].trim()
+    return { type: className ? `class ${className}` : 'class' }
+  }
+}
+
+function toString (str, maxLength) {
+  const size = str.length
+
+  if (size <= maxLength) {
+    return { type: 'string', value: str }
+  }
+
+  return {
+    type: 'string',
+    value: str.substr(0, maxLength),
+    truncated: true,
+    size
+  }
+}
+
+function toObject (type, props, maxLength) {
+  if (props === undefined) return notCaptured(type)
+  return { type, fields: processProperties(props, maxLength) }
+}
+
+function toArray (type, elements, maxLength) {
+  if (elements === undefined) return notCaptured(type)
+
+  // Perf: Create array of expected size in advance (expect that it contains only one non-enumrable element)
+  const expectedLength = elements.length - 1
+  const result = { type, elements: new Array(expectedLength) }
+
+  let i = 0
+  for (const elm of elements) {
```

Comment:
cap at collectionSize limit

Reply:
That's out of scope for this PR (see description) and will be added in an upcoming PR.